File: backend/src/main/kotlin/ase/athlete_view/domain/user/pojo/dto/PreferencesDTO.kt
package ase.athlete_view.domain.user.pojo.dto

import ase.athlete_view.domain.user.pojo.entity.NotificationPreferenceType
import ase.athlete_view.domain.user.pojo.entity.Preferences

class PreferencesDTO (
    var emailNotifications: Boolean,
    var commentNotifications: NotificationPreferenceType,
    var ratingNotifications: NotificationPreferenceType,
    var otherNotifications: NotificationPreferenceType,
    var shareHealthWithTrainer: Boolean

    ) {
    fun toEntity(id: Long?): Preferences {
        return Preferences(id, emailNotifications, commentNotifications, ratingNotifications, otherNotifications, shareHealthWithTrainer)
    }

    override fun toString(): String {
        return "PreferencesDTO(emailNotifications=$emailNotifications, commentNotifications=$commentNotifications, ratingNotifications=$ratingNotifications, otherNotifications=$otherNotifications, shareHealthWithTrainer=$shareHealthWithTrainer)"
    }

}

File: frontend/src/app/domain/user/dto/preferences-dto.ts
export interface PreferencesDto {
  emailNotifications: boolean,
  commentNotifications: PreferenceNotificationType,
  ratingNotifications: PreferenceNotificationType,
  otherNotifications: PreferenceNotificationType,
  shareHealthWithTrainer: boolean,
}


export enum PreferenceNotificationType {
  EMAIL = "EMAIL",
  PUSH = "PUSH",
  NONE = "NONE",
  BOTH = "BOTH",
}

export const PreferenceNotificationMapper = new Map<PreferenceNotificationType, string>([
  [PreferenceNotificationType.EMAIL, 'Email'],
  [PreferenceNotificationType.PUSH, 'Push'],
  [PreferenceNotificationType.NONE, 'None'],
  [PreferenceNotificationType.BOTH, 'Email & Push'],
]);
